Wrong error diagnostics message 'Cannot find module xxxx'

**TypeScript Version:**  3.4.0-dev.20190315

**Search Terms:**

`Cannot find module`, `wrong message`

**Steps to reproduce**

1. Create a new workspace (empty folder). Create subfolder `test-plugin` in it.

2. Install `npm install -g yo @theia/generator-plugin`

3. Generate a frontend plug-in: `yo @theia/plugin`. So, the folders structure should be as following:

```
ws-root
 |- test-plugin
    |- dist
    |- src
    |  |- test-plugin-frontend.ts
    |- node_modules   
    |  |- ...
    |- package.json
    |- tsconfig.json
    |- ...
```

4. Open `test-plugin-frontend.ts` and paste the following content:

```typescript
import * as theia from '@theia/plugin';

export function start(context: theia.PluginContext) {
    theia.window.onDidChangeVisibleTextEditors((editors: theia.TextEditor[]) => {
        let list: string = '';
        for (let editor of editors) {
            list += editor.document.uri.toString() + ', ';
        }
        console.log(list);
    });
  
   // theia.window.onDid
}

export function stop() { }
```
5. Build the project using `yarn` command

6. Uncomment the commented line. Change a few symbols and comment it again (Use `Ctrl + /`shortcut). Save the editor content.

**Expected behavior:**

No error diagnostic message, nothing has changed actually.

**Actual behavior:**

Error diagnostic message is appeared on the `@theia/plugin` import: `[typescript] Cannot find module '@theia/plugin'`

<img width="1615" alt="Screen Shot 2019-03-14 at 15 18 04" src="https://user-images.githubusercontent.com/3082655/54364022-65cf2f80-466c-11e9-848a-7dbe27960a7b.png">

--- a/src/compiler/program.ts
+++ b/src/compiler/program.ts
@@ -1134,7 +1134,7 @@ namespace ts {
             function moduleNameResolvesToAmbientModuleInNonModifiedFile(moduleName: string): boolean {
                 const resolutionToFile = getResolvedModule(oldSourceFile!, moduleName);
                 const resolvedFile = resolutionToFile && oldProgram!.getSourceFile(resolutionToFile.resolvedFileName);
-                if (resolutionToFile && resolvedFile && !resolvedFile.externalModuleIndicator) {
+                if (resolutionToFile && resolvedFile) {
                     // In the old program, we resolved to an ambient module that was in the same
                     //   place as we expected to find an actual module file.
                     // We actually need to return 'false' here even though this seems like a 'true' case

--- a/src/server/project.ts
+++ b/src/server/project.ts
@@ -958,6 +958,9 @@ namespace ts.server {
             );
             const elapsed = timestamp() - start;
             this.writeLog(`Finishing updateGraphWorker: Project: ${this.getProjectName()} Version: ${this.getProjectVersion()} structureChanged: ${hasNewProgram} Elapsed: ${elapsed}ms`);
+            if (this.program !== oldProgram) {
+                this.print();
+            }
             return hasNewProgram;
         }
 

--- a/src/testRunner/unittests/tsserver/projectErrors.ts
+++ b/src/testRunner/unittests/tsserver/projectErrors.ts
@@ -479,6 +479,90 @@ namespace ts.projectSystem {
                 session.clearMessages();
             }
         });
+
+        it("Correct errors when resolution resolves to file that has same ambient module and is also module", () => {
+            const projectRootPath = "/users/username/projects/myproject";
+            const aFile: File = {
+                path: `${projectRootPath}/src/a.ts`,
+                content: `import * as myModule from "@custom/plugin";
+function foo() {
+  // hello
+}`
+            };
+            const config: File = {
+                path: `${projectRootPath}/tsconfig.json`,
+                content: JSON.stringify({ include: ["src"] })
+            };
+            const plugin: File = {
+                path: `${projectRootPath}/node_modules/@custom/plugin/index.d.ts`,
+                content: `import './proposed';
+declare module '@custom/plugin' {
+    export const version: string;
+}`
+            };
+            const pluginProposed: File = {
+                path: `${projectRootPath}/node_modules/@custom/plugin/proposed.d.ts`,
+                content: `declare module '@custom/plugin' {
+    export const bar = 10;
+}`
+            };
+            const files = [libFile, aFile, config, plugin, pluginProposed];
+            const host = createServerHost(files);
+            const session = createSession(host, { canUseEvents: true });
+            const service = session.getProjectService();
+            openFilesForSession([aFile], session);
+
+            checkNumberOfProjects(service, { configuredProjects: 1 });
+            session.clearMessages();
+            checkErrors();
+
+            session.executeCommandSeq<protocol.ChangeRequest>({
+                command: protocol.CommandTypes.Change,
+                arguments: {
+                    file: aFile.path,
+                    line: 3,
+                    offset: 8,
+                    endLine: 3,
+                    endOffset: 8,
+                    insertString: "o"
+                }
+            });
+            checkErrors();
+
+            function checkErrors() {
+                host.checkTimeoutQueueLength(0);
+                const expectedSequenceId = session.getNextSeq();
+                session.executeCommandSeq<protocol.GeterrRequest>({
+                    command: server.CommandNames.Geterr,
+                    arguments: {
+                        delay: 0,
+                        files: [aFile.path],
+                    }
+                });
+
+                host.checkTimeoutQueueLengthAndRun(1);
+
+                checkErrorMessage(session, "syntaxDiag", { file: aFile.path, diagnostics: [] }, /*isMostRecent*/ true);
+                session.clearMessages();
+
+                host.runQueuedImmediateCallbacks(1);
+
+                checkErrorMessage(session, "semanticDiag", { file: aFile.path, diagnostics: [] });
+                session.clearMessages();
+
+                host.runQueuedImmediateCallbacks(1);
+
+                checkErrorMessage(session, "suggestionDiag", {
+                    file: aFile.path,
+                    diagnostics: [
+                        createDiagnostic({ line: 1, offset: 1 }, { line: 1, offset: 44 }, Diagnostics._0_is_declared_but_its_value_is_never_read, ["myModule"], "suggestion", /*reportsUnnecessary*/ true),
+                        createDiagnostic({ line: 2, offset: 10 }, { line: 2, offset: 13 }, Diagnostics._0_is_declared_but_its_value_is_never_read, ["foo"], "suggestion", /*reportsUnnecessary*/ true)
+                    ],
+                });
+                checkCompleteEvent(session, 2, expectedSequenceId);
+                session.clearMessages();
+            }
+        });
     });
 
     describe("unittests:: tsserver:: Project Errors for Configure file diagnostics events", () => {

